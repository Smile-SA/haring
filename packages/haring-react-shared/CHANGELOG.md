# @smile/haring-react-shared

## 0.15.0

### Minor Changes

- [#194](https://github.com/Smile-SA/haring/pull/194) [`97a610a`](https://github.com/Smile-SA/haring/commit/97a610aa02b86423b723ef8ae5d45e8ecf346f92) Thanks [@vapersmile](https://github.com/vapersmile)! - Upgrade mantine version to 7.11.x

## 0.14.0

## 0.13.0

### Minor Changes

- [#178](https://github.com/Smile-SA/haring/pull/178) [`ce8f39a2c52590b2c91950450a84222434913743`](https://github.com/Smile-SA/haring/commit/ce8f39a2c52590b2c91950450a84222434913743) Thanks [@tonai](https://github.com/tonai)! - Rename packages

## 0.12.0

## 0.11.0

## 0.10.0

### Minor Changes

- [#149](https://github.com/Smile-SA/haring/pull/149) [`2706a09`](https://github.com/Smile-SA/haring/commit/2706a097fd86b072cf21d12fe2a97427883671cf) Thanks [@vapersmile](https://github.com/vapersmile)! - Add Calendar style on globlaStyle and add a preview

- [#150](https://github.com/Smile-SA/haring/pull/150) [`34fabfb`](https://github.com/Smile-SA/haring/commit/34fabfbd1a2c048c39adc567548a3ee7e85074ee) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Fixed missing global styles for focus-visible color

## 0.9.0

### Minor Changes

- [`d430e2c`](https://github.com/Smile-SA/haring/commit/d430e2cbcd26871c4db0c53f92a4e4baa6af839c) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Migrated from Mantine V6 to V7

## 0.8.1

## 0.8.0

### Minor Changes

- [#108](https://github.com/Smile-SA/haring/pull/108) [`320634d`](https://github.com/Smile-SA/haring/commit/320634d7ec2048a328cfda1dbcff5d1b70a32563) Thanks [@vapersmile](https://github.com/vapersmile)! - Add custom default style for carousel mantine composent and add 2-Mantine/carousel page

## 0.7.0

### Minor Changes

- [#87](https://github.com/Smile-SA/haring/pull/87) [`dbc06ca`](https://github.com/Smile-SA/haring/commit/dbc06ca55961b69663ab7fdc02609c6525ae361d) Thanks [@tonai](https://github.com/tonai)! - Add theme provider and update barrel file for better next.js compatibility

- [#95](https://github.com/Smile-SA/haring/pull/95) [`5696400`](https://github.com/Smile-SA/haring/commit/5696400e6f703da52db5f7199b50f8251fa76726) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `FilterList` component, renamed `FiltersBar` in `SidebarFilters`
  and `FiltersCheckboxList` in `SearchableCheckboxList`,
  refactored `SearchableCheckboxList`, added common `IFilter` type in shared
  package, added topContent prop to `SwitchableView`, fixed `BrowsingPage` with
  name changes.

## 0.6.0

### Minor Changes

- [#74](https://github.com/Smile-SA/haring/pull/74) [`0ad5377`](https://github.com/Smile-SA/haring/commit/0ad5377535e7c19941da301e8a2ee7298ab70f91) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Refactored actions in `ThumbnailGrid` and `TableGridView` components, moved `typeGuard` helper to shared package, added `typeGuardInterface` function, added tests

## 0.5.0

### Minor Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add dynamic componentProps property in actions

- [#65](https://github.com/Smile-SA/haring/pull/65) [`bd9cfe4`](https://github.com/Smile-SA/haring/commit/bd9cfe42d0b22f5f7f5e7b0de30fdfb22ad3e1c8) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Reworked action/confirm types and moved then in `react-front-kit-shared` then refactored types in `Table`, `Thumbnail` and `ConfirmModal` , Added `ThumbnailGrid` and `TableGridView` components,

- [#68](https://github.com/Smile-SA/haring/pull/68) [`c8b5a39`](https://github.com/Smile-SA/haring/commit/c8b5a3978c8ce7133da0b498d9f0b326f07eb737) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `storybook-pages` package containing example pages, renamed `TestPage`
  into `BrowsingPage` and completed it with integration of `TableGridView`, fixed
  various component styles and props related to page.

### Patch Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add server barrel files

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/haring/pull/53) [`01ee3ea`](https://github.com/Smile-SA/haring/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
