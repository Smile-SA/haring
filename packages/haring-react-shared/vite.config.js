import { resolve } from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.tsx'),
      fileName: '[name]',
      formats: ['cjs', 'es'],
    },
    rollupOptions: {
      external: [
        '@mantine/carousel',
        '@mantine/core',
        '@mantine/dates',
        '@mantine/dropzone',
        '@mantine/hooks',
        '@phosphor-icons/react',
        '@smile/haring-react',
        '@smile/haring-react-dates',
        '@smile/haring-react-dropzone',
        '@smile/haring-react-shared',
        '@smile/haring-react-table',
        '@storybook/addon-actions',
        '@storybook/preview-api',
        '@tabler/icons-react',
        '@testing-library/react',
        'dayjs',
        'embla-carousel-react',
        'mantine-react-table',
        'react',
        'react/jsx-runtime',
      ],
      input: {
        index: resolve(__dirname, 'src/index.tsx'),
        server: resolve(__dirname, 'src/server.tsx'),
        'storybook-utils': resolve(__dirname, 'src/storybook-utils/index.ts'),
        'test-utils': resolve(__dirname, 'src/test-utils/index.ts'),
      },
      output: {
        banner: (chunkInfo) => {
          if (
            ['src/index.tsx'].find((modulePath) =>
              chunkInfo.facadeModuleId?.endsWith(modulePath),
            )
          ) {
            return `"use client"`;
          }
          return '';
        },
        preserveModules: false,
      },
    },
  },
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
      tsconfigPath: './tsconfig.build.json',
    }),
  ],
});
