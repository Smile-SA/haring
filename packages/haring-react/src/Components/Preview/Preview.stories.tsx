import type { Meta, StoryObj } from '@storybook/react';

import { Preview as Cmp } from './Preview';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/Preview',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const Pdf: IStory = {
  args: {
    url: './example.pdf',
  },
};

export const ImagePng: IStory = {
  args: {
    url: './image.png',
  },
};

export const ImageSvg: IStory = {
  args: {
    url: './logo.svg',
  },
};

export const Video: IStory = {
  args: {
    url: './video.mp4',
  },
};

export const Audio: IStory = {
  args: {
    url: './audio.mp3',
  },
};
