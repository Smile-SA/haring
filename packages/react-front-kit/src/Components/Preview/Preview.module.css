.audio {
  max-width: 100%;
}

.iframe {
  background: var(--mantine-color-white);
  box-shadow:
    0 3.4348926544189453px 2.7479140758514404px 0 rgba(0, 0, 0, 0.0155),
    0 8.687101364135742px 6.949680805206299px 0 rgba(0, 0, 0, 0.0222),
    0 17.720870971679688px 14.176697731018066px 0 rgba(0, 0, 0, 0.0278),
    0 36.501644134521484px 29.201316833496094px 0 rgba(0, 0, 0, 0.0345),
    0 100px 80px 0 rgba(0, 0, 0, 0.05);
  width: 100%;
}

.image {
  max-width: 100%;
}

.root {
  text-align: center;
  width: 100%;
}

.video {
  max-width: 100%;
}
