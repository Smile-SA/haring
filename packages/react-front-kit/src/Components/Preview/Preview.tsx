'use client';

import type { ReactElement } from 'react';

import { useEffect, useState } from 'react';

import classes from './Preview.module.css';

export interface IPreviewProps {
  defaultAspectRatio?: number;
  defaultHeight?: number;
  defaultMimeType?: string;
  defaultWidth?: number;
  url: string;
}

export function Preview(
  props: IPreviewProps &
    JSX.IntrinsicElements['audio' | 'iframe' | 'img' | 'video'],
): ReactElement {
  const {
    defaultAspectRatio,
    defaultHeight,
    defaultWidth,
    defaultMimeType,
    url,
    ...restProps
  } = props;
  const [type, setType] = useState(defaultMimeType);
  const [width, setWidth] = useState<number | undefined>(
    defaultWidth ??
      (defaultAspectRatio && defaultHeight
        ? defaultAspectRatio / defaultHeight
        : undefined),
  );
  const [height, setHeight] = useState<number | undefined>(
    defaultHeight ??
      (defaultAspectRatio && defaultWidth
        ? defaultAspectRatio * defaultWidth
        : undefined),
  );

  const mimeType = defaultMimeType ?? type;
  const isImage = mimeType?.includes('image/');
  const isVideo = mimeType?.includes('video/');
  const isAudio = mimeType?.includes('audio/');
  const isOther = mimeType && !isImage && !isVideo && !isAudio;

  useEffect(() => {
    if (!defaultMimeType && url) {
      fetch(url)
        .then((r) => r.blob())
        .then((blob) => setType(blob.type))
        .catch((error) => console.error(error));
    }
  }, [defaultMimeType, url]);

  useEffect(() => {
    if (!defaultAspectRatio && isImage && url) {
      const img = new Image();
      img.onload = () => {
        setWidth(img.width);
        setHeight(img.height);
      };
      img.src = url;
    }
  }, [defaultAspectRatio, isImage, url]);

  return (
    <div className={classes.root}>
      {Boolean(isImage && width && height) && (
        <img
          alt=""
          className={classes.image}
          height={height}
          src={url}
          width={width}
          {...(restProps as JSX.IntrinsicElements['img'])}
        />
      )}
      {Boolean(isVideo) && (
        // eslint-disable-next-line jsx-a11y/media-has-caption
        <video
          className={classes.video}
          controls
          src={url}
          style={{ aspectRatio: defaultAspectRatio }}
          {...(restProps as JSX.IntrinsicElements['video'])}
        />
      )}
      {Boolean(isAudio) && (
        // eslint-disable-next-line jsx-a11y/media-has-caption
        <audio
          className={classes.audio}
          controls
          src={url}
          {...(restProps as JSX.IntrinsicElements['audio'])}
        />
      )}
      {Boolean(isOther) && (
        // eslint-disable-next-line react/iframe-missing-sandbox
        <iframe
          className={classes.iframe}
          src={url}
          style={{ aspectRatio: defaultAspectRatio ?? '3/4' }}
          title="Preview"
          {...(restProps as JSX.IntrinsicElements['iframe'])}
        />
      )}
    </div>
  );
}
