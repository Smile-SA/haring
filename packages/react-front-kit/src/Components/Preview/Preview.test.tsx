import { renderWithProviders } from '@smile/haring-react-shared/test-utils';

import { Preview } from './Preview';

describe('Preview', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <Preview defaultMimeType="image/png" url="./image.png" />,
    );
    expect(container).toMatchSnapshot();
  });
});
