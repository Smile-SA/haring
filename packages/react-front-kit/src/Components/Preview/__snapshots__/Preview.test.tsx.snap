// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview matches snapshot 1`] = `
<div>
  <div
    class="mantine-vuwiwp"
  />
</div>
`;
